src/api.rs

```diff
@@ -44,28 +44,38 @@ pub type Error = errno::Errno;
 /// Simpler `Result` type with the error already set.
 pub type Result<T> = result::Result<T, Error>;
 
-fn check_call<T>(res: libc::c_long, value: T) -> Result<T> {
-    if res == -1 {
-        Err(errno::errno())
-    } else {
-        Ok(value)
+fn check_call(res: libc::c_long) -> Result<()> {
+    match res {
+        -1 => Err(errno::errno()),
+        _ => Ok(()),
     }
 }
 
-fn check_call_ret(res: libc::c_long) -> Result<libc::c_long> {
-    if res == -1 {
-        Err(errno::errno())
-    } else {
-        Ok(res)
-    }
+fn check_call_key(res: KeyringSerial) -> Result<Key> {
+    check_call(res.get().into())?;
+    Ok(Key::new_impl(res))
 }
 
-fn check_call_ret_serial(res: KeyringSerial) -> Result<KeyringSerial> {
-    if res == -1 {
-        Err(errno::errno())
-    } else {
-        Ok(res)
-    }
+fn check_call_keyring(res: KeyringSerial) -> Result<Keyring> {
+    check_call(res.get().into())?;
+    Ok(Keyring::new_impl(res))
+}
+
+fn into_serial(res: libc::c_long) -> KeyringSerial {
+    KeyringSerial::new(res as i32).unwrap()
+}
+
+fn request_impl<K: KeyType>(
+    description: &str,
+    info: Option<&str>,
+    id: Option<KeyringSerial>,
+) -> KeyringSerial {
+    let type_cstr = CString::new(K::name()).unwrap();
```

Comment:
Maybe I wasn't doing this when I first wrote this code, but I've found `expect` to be much better for tracking down when these things go wrong. However, looking at how many there are just in the context here, this can be done in a future PR to avoid making this one noisier.

Reply:
I think the best way to deal with this would be to return an error, but I think it’s best addressed comprehensively across the project. I’ve opened #30 to discuss this.